sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/SingleRowScanConverter.java

```diff
@@ -36,6 +42,22 @@ public boolean canConvert(ResolvedSingleRowScan zetaNode) {
 
   @Override
   public RelNode convert(ResolvedSingleRowScan zetaNode, List<RelNode> inputs) {
-    return LogicalValues.createOneRow(getCluster());
+    return createOneRow(getCluster());
+  }
+
+  // This function is a reimplementation of Calcite's LogicalValues.createOneRow() with a single
+  // line change: SqlTypeName.INTEGER replaced by SqlTypeName.BIGINT.
+  // Would like to use LogicalValues.createOneRow(), but it uses type SqlTypeName.INTEGER which
+  // correspond to TypeKind.TYPE_INT32 in ZetaSQL, a type not supported in PRODUCT_EXTERNAL mode.
+  private static LogicalValues createOneRow(RelOptCluster cluster) {
```

Comment:
I don't understand this. More context please?

Reply:
Added more context in the comments. Please let me know if this makes sense.